haystack/nodes/doc_language_classifier/transformers.py

```diff
@@ -0,0 +1,176 @@
+            task="text-classification",
+            model=model_name_or_path,
+            tokenizer=tokenizer,
+            device=resolved_devices[0],
+            revision=model_version,
+            top_k=1,
+            use_auth_token=use_auth_token,
+        )
+        self.batch_size = batch_size
+        self.progress_bar = progress_bar
+
+    def predict(self, documents: List[Document], batch_size: Optional[int] = None) -> List[Document]:
+        """
+        Detect the languge of Documents and add the output to the Documents metadata.
+        :param documents: list of Documents to detect language.
+        :param batch_size: The number of Documents to classify at a time.
+        :return: List of Documents, where Document.meta["language"] contains the predicted language
+        """
+        if len(documents) == 0:
+            raise AttributeError(
+                "TransformersDocumentLanguageClassifier needs at least one document to predict the language."
+            )
+        if batch_size is None:
+            batch_size = self.batch_size
+
+        texts = [doc.content for doc in documents]
+        batches = self._get_batches(texts, batch_size=batch_size)
+        predictions = []
+        pb = tqdm(total=len(texts), disable=not self.progress_bar, desc="Predicting the language of documents")
+        for batch in batches:
+            batched_prediction = self.model(batch, top_k=1, truncation=True)
+            predictions.extend(batched_prediction)
+            pb.update(len(batch))
+        pb.close()
+        for prediction, doc in zip(predictions, documents):
+            doc.meta["language"] = prediction[0]["label"]
+        return documents
+
+    def predict_batch(self, documents: List[List[Document]], batch_size: Optional[int] = None) -> List[List[Document]]:
+        """
+        Detect the documents language and add the output to the document's meta data.
+        :param documents: list of lists of Documents to detect language.
+        :return: List of lists of Documents, where Document.meta["language"] contains the predicted language
+        """
+        if len(documents) == 0 or all(len(docs_list) == 0 for docs_list in documents):
+            raise AttributeError(
+                "TransformersDocumentLanguageClassifier needs at least one document to predict the language."
+            )
+        if batch_size is None:
+            batch_size = self.batch_size
+
+        number_of_documents = [len(doc_list) for doc_list in documents]
+        flattened_documents = list(itertools.chain.from_iterable(documents))
+        docs_with_preds = self.predict(flattened_documents, batch_size=batch_size)
+
+        # Group documents together
+        grouped_documents = []
+        left_idx = 0
+        right_idx = 0
+        for number in number_of_documents:
+            right_idx = left_idx + number
+            grouped_documents.append(docs_with_preds[left_idx:right_idx])
+            left_idx = right_idx
```

Comment:
I think you can make it work without indices:

```suggestion
        grouped_documents = []
        for documents_group_length in number_of_documents:
            grouped_documents.append(docs_with_preds[:documents_group_length])
            docs_with_preds = docs_with_preds[documents_group_length:]
```

(untested)

Reply:
Makes sense...
Let's see if I can do better :smiley: